packages/presentation/lib/base/bloc.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import '../navigation/app_navigator.dart';
+import 'bloc_data.dart';
+
+abstract class Bloc {
+  Stream<BlocData> get dataStream;
+
+  void initState();
```

Comment:
I think you need add dispose method to bloc interface, for later dispose stream controller to prevent memory leaks.